### PR TITLE
Add P2P reference information to deps.json.

### LINKED ISSUE #121: Scenario: enable P2P references across all aspects of the sdk functionality
- [ ] ensure P2P references are written to the .deps.json file.  (Do they get put into the lock file by NuGet? We currently only use lock file information for .deps.json generation).
- [ ] ensure they are copied to the build and publish folders
- [ ] ensure P2P references' resource assemblies are copied to the build and publish folders.  And also that the P2P references' resource assemblies are added to the .deps.json file.

### PR DESCRIPTION
**Pull request by eerhardt**
Fix #121

@nguerrera @dsplaisted 

/cc @dotnet/project-system

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs [hunk 1]
```diff
@@ -34,12 +42,58 @@ public static SingleProjectInfo Create(string projectPath, string name, string v
                 resourceAssemblies.Add(new ResourceAssemblyInfo(culture, relativePath));
             }
 
-            return new SingleProjectInfo(projectPath, name, version, resourceAssemblies);
+            return new SingleProjectInfo(projectPath, name, version, $"{name}.dll", resourceAssemblies);
```
Comment (eerhardt) [inline src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs]: I logged #323 to fix this hard-coded extension.

File: src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs [hunk 2]
```diff
@@ -34,12 +42,58 @@ public static SingleProjectInfo Create(string projectPath, string name, string v
...
-            return $"{Name}.dll";
+            Dictionary<string, SingleProjectInfo> projectReferences = new Dictionary<string, SingleProjectInfo>();
+
+            foreach (ITaskItem projectReferencePath in projectReferencePaths)
+            {
+                string sourceProjectFile = projectReferencePath.GetMetadata("MSBuildSourceProjectFile");
+
+                if (string.IsNullOrEmpty(sourceProjectFile))
+                {
+                    throw new Exception($"Could not find valid 'MSBuildSourceProjectFile' metadata on ReferencePath '{projectReferencePath.ItemSpec}'");
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs]: ReportUserErrorException (or whatever we called it)?

File: src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs [hunk 3]
```diff
@@ -34,12 +42,58 @@ public static SingleProjectInfo Create(string projectPath, string name, string v
...
+                    sourceProjectFile,
+                    new SingleProjectInfo(sourceProjectFile, name, version, outputName, resourceAssemblies));
+            }
+
+            foreach (ITaskItem projectReferenceSatellitePath in projectReferenceSatellitePaths)
+            {
+                string sourceProjectFile = projectReferenceSatellitePath.GetMetadata("MSBuildSourceProjectFile");
+
+                if (string.IsNullOrEmpty(sourceProjectFile))
+                {
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs]: Same question

**Review by nguerrera — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs (hunk 2 above)
Comment (eerhardt) [inline src/Tasks/Microsoft.NET.Build.Tasks/SingleProjectInfo.cs]: Should we do something else here?  Like skip it?  It feels like that will just cause other issues downstream...

Maybe log a warning?

**Pull request merged**